ls1intum/Artemis
# `Exam mode`: Improve wording for grades with bonus

### Checklist
#### General


#### Client

### Motivation and Context


During a testing session for https://github.com/ls1intum/Artemis/pull/5668, other developers mentioned that the explanations for exam results can be confusing for the students. This PR aims to make the distinction between achieved, max and bonus points clearer. Closes https://github.com/ls1intum/Artemis/issues/5188.

### Description

Added a footer row to display the total values for "Your Points", "Achieved normal points", "Achieved bonus points" columns. Improved explanatory text for exam grade and bonus. Removed redundant whitespace in the exam results mentioned in https://github.com/ls1intum/Artemis/issues/5188.

### Steps for Testing

Prerequisites:
- 1 graded Exam without any bonus exercises or bonus configuration
- 1 graded Exam with bonus exercise(s) but without bonus course/exam
- 1 graded Exam with bonus exercise(s) and bonus course/exam

1. Log in to Artemis

For each prerequisite exam above:

2. Go to the student view of exam summary either with an instructor or a student.
3. Check that the "Your Points", "Achieved normal points", "Achieved bonus points" columns are summed correctly in the "Total" footer row in the summary table at the top.
4. Check the explanatory texts about grade and points (the one starting with "You achieved...") seems correct and clear.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|
| exam-points-summary.component.ts | 79.16% | 100% |

### Screenshots



![image](https://user-images.githubusercontent.com/18427209/199285734-94b93518-9a78-47c1-8ee5-8cad9b6c639a.png)
